tests/system/python/api/test_browser_assets.py

```diff
@@ -0,0 +1,317 @@
+    ])
+    def test_get_asset_query_bad_data(self, foglamp_url, request_params, response_code, response_message):
+        conn = http.client.HTTPConnection(foglamp_url)
+        conn.request("GET", '/foglamp/asset/{}{}'.format(ASSET_NAME, request_params))
+        r = conn.getresponse()
+        conn.close()
+        assert response_code == r.status
+        assert response_message == r.reason
+
+    def test_get_asset_reading(self, foglamp_url):
+        conn = http.client.HTTPConnection(foglamp_url)
+        conn.request("GET", '/foglamp/asset/{}/{}'.format(ASSET_NAME, SENSOR))
+        r = conn.getresponse()
+        assert 200 == r.status
+        r = r.read().decode()
+        jdoc = json.loads(r)
+        assert len(jdoc), "No data found"
+        i = 0
+        for val in SENSOR_VALUES:
+            assert val == jdoc[i][SENSOR]
+            i += 1
+
+    @pytest.mark.parametrize(("query", "expected_count", "expected_values"), [
+        ('?limit=1', 1, [SENSOR_VALUES[0]]),
+        ('?limit=1&skip=1', 1, [SENSOR_VALUES[1]]),
+        ('?seconds=59', 2, SENSOR_VALUES[0:2]),
+        ('?minutes=15', 4, SENSOR_VALUES[0:4]),
+        ('?hours=4', 5, SENSOR_VALUES[0:5]),
+        ('?hours=20&minutes=20&seconds=59&limit=20', 2, SENSOR_VALUES[0:2]), # Verify that if a combination of hrs, min, sec is used, shortest period will apply
+        ('?limit=&hours=&minutes=&seconds=', 6, SENSOR_VALUES)
+        # In case of empty params, all values are returned
+    ])
+    def test_get_asset_readings_query(self, foglamp_url, query, expected_count, expected_values):
+        conn = http.client.HTTPConnection(foglamp_url)
+        conn.request("GET", '/foglamp/asset/{}/{}{}'.format(ASSET_NAME, SENSOR, query))
+        r = conn.getresponse()
+        assert 200 == r.status
+        r = r.read().decode()
+        jdoc = json.loads(r)
+        assert len(jdoc) == expected_count
+        i = 0
+        for item in expected_values:
+            assert item == jdoc[i][SENSOR]
+            i += 1
+
+    def test_get_asset_summary(self, foglamp_url):
+        conn = http.client.HTTPConnection(foglamp_url)
+        conn.request("GET", '/foglamp/asset/{}/summary'.format(ASSET_NAME))
+        r = conn.getresponse()
+        assert 200 == r.status
+        r = r.read().decode()
+        jdoc = json.loads(r)
+        assert len(jdoc), "No data found"
+        summary = jdoc[0][SENSOR]
+        avg = sum(SENSOR_VALUES) / len(SENSOR_VALUES)
+        assert avg == summary['average']
+        assert max(SENSOR_VALUES) == summary['max']
+        assert min(SENSOR_VALUES) == summary['min']
+
+    def test_get_asset_readings_summary_invalid_sensor(self, foglamp_url):
+        conn = http.client.HTTPConnection(foglamp_url)
+        invalid_sensor = "invalid"
+        conn.request("GET", '/foglamp/asset/{}/{}/summary'.format(ASSET_NAME, invalid_sensor))
+        r = conn.getresponse()
+        assert 200 == r.status
```

Comment:
why this is a 200; should not it be 404?

Reply:
Architect decision, please see the [discussion](https://scaledb.atlassian.net/browse/FOGL-1292?focusedCommentId=39059&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-39059)